.circleci/config.yml

```diff
@@ -660,6 +660,18 @@ jobs:
       TEST_MODE: << parameters.test_mode >>
     parallelism: << parameters.parallel_num >>
     steps:
+      - checkout-code
+      # When running on PRs, only UI changes are exercised.
+      - when:
+          condition:
+            equal: [ "local", << parameters.env_name >> ]
+          steps:
+            - halt-test-check:
+                dir_names: "ui"
```

Comment:
I might be reading this wrong, but it seems like there's a mismatch between `When running on PRs` and `equal: [ "local", << parameters.env_name >> ]`.

Reply:
Correct. It is a coincidence, and that is not good. However, since this should have a pretty significant impact on PR throughput, I judge that it is worth the cost. If things seem to work out once this is merged, we'll have an easier time making improvements.

Reply:
Sorry, I'm not sure what you mean.

My understanding of the intent of this change: "When running this in **PRs**, only run this task when UI files have changed."

My understanding of the configuration: "When running this **locally**, only run this task when UI files have changed."

Either one of the above statements is wrong, or the configuration is incorrect. Which is it?

Reply:
Local means local to Circle (run stuff on the build machine) rather than in the test, staging, etc. environment. The only time we have to queue is when we're going to do something on a shared environment that may have contention. Is that helpful?

Reply:
Yes, that makes sense!